### ISSUE TITLE
[a11y]: [Windows High Contrast Mode]: [IconButton] The icon is not visible in high contrast mode

### ISSUE THREAD

**Issue by nkn2022** [carbon-design-system/carbon#17147]
### Package

@carbon/react

### Browser

Chrome

### Operating System

Windows

### Package version

1.57.0

### React version

17.0.1

### Automated testing tool and ruleset

Manual Verification

### Description

Manual verification of the above requirement brought up this issue. 
Ref: 
Requirement - [1.4.5 Images of Text](https://www.ibm.com/able/requirements/requirements/?version=7_2#1_4_5)
Manual Verification - [Check use of color, contrast, and images of text](https://www.ibm.com/able/toolkit/verify/manual/#color)

### WCAG 2.1 Violation

1.4.5 Images of Text

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-iconbutton--playground&args=kind:ghost

### Steps to reproduce

Go to Carbon storybook URL - https://react.carbondesignsystem.com/?path=/story/components-iconbutton--playground&args=kind:ghost

<img width="755" alt="image" src="https://github.com/user-attachments/assets/6baa4353-5aa0-46ba-a83d-20d3afa8760c">

You can see that the icon is not visible. 

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
This has come up a few times before

* https://github.com/carbon-design-system/carbon/issues/11557
* https://github.com/carbon-design-system/carbon/issues/9934

I'm not sure what's caused this latest regression

**Comment by ziyadzulfikar** [comment]
Hi @tay1orjones,

If this issue is still open, I’d like to work on it. Please assign it to me. Thanks!

**Comment by nkn2022** [comment]
Any update on this?

**Comment by ziyadzulfikar** [comment]
If this PR is assigned to me, then I can have a look on this.

**Comment by alexandravsmith** [comment]
Hey, could I get this issue assigned to me?

**Comment by JuanMacias153** [comment]
Hey, if this issue is still open, could I be assigned to it?

**Comment by JoseG2003-g** [comment]
Can we still contribute to this issue?

**Comment by nkn2022** [comment]
Any update on this? 
If any of you have a fix, you should be able to open a PR? Is that not allowed?

@tay1orjones Please help comment on this?

**Comment by nkn2022** [comment]
Thank you!
